### ISSUE TITLE
console_poller_unregister: Assertion `i < console->n_pollers' failed.

### ISSUE THREAD

**Issue by shenki** [openbmc/obmc-console#12]
When testing with an experimental 4.12 based kernel tree:

```
obmc-console-server: Can't open log buffer file /home/joel/tmp/obmc/var/log/obmc-console.log: No such file or directory
obmc-console-server: failed writing to local tty; disabling: Bad file descriptor
obmc-console-server: console-server.c:372: console_poller_unregister: Assertion `i < console->n_pollers' failed.
3 handlers
  log [inactive]
  socket [active]
Set ttyS0 terminal baud rate to 115200
Set ttyS0 for raw byte handling
  tty [active]
```

**Comment by shenki** [comment]
I guess the first one is because I built locally.

**Comment by shenki** [comment]
If I remove the uart mirroring entries from /etc/obmc-console.conf, the daemon behaves normally.

**Comment by jk-ozlabs** [comment]
Taking a look.

**Comment by jk-ozlabs** [comment]
@shenki: that first line is a bit suspicious: are you running this inside a BMC system? or elsewhere? or with an out-of-yocto build of obmc-console?

**Comment by shenki** [comment]
@jk-ozlabs I built the binary locally, and then copied over to the bmc

I did this so I could have symbols for the backtrace I was getting.

**Comment by jk-ozlabs** [comment]
Fixed in 67eab046047.
